### PR TITLE
More 1.12 compat

### PR DESCRIPTION
**Pull request by Keno**
With this and the LCU fixes, Revise should pass tests again on 1.12.

### CODE REVIEW

File: src/loading.jl [hunk 1]
```diff
@@ -29,13 +36,17 @@ function parse_pkg_files(id::PkgId)
...
     if use_compiled_modules()
-        cachefile, includes, reqs = pkg_fileinfo(id)
+        cachefile, includes, reqs, buildid = pkg_fileinfo(id)
         if cachefile !== nothing
             @assert includes !== nothing
             @assert reqs !== nothing
             pkgdata.requirements = reqs
             for chi in includes
-                mod = Base.root_module(id)
+                if VERSION >= v"1.12.0-DEV.764" && haskey(Base.loaded_precompiles, id => buildid)
```
Comment (Keno) [inline src/loading.jl:45]: @vtjnash is this the expected fix for this?

Comment (vtjnash) [reply src/loading.jl:46]: That probably depends upon what the code is attempting to do. It is supposed to be buildid specific or not?

Comment (Keno) [reply src/loading.jl:46]: This code is adding the files that a .ji file depends on to Revise's watch list, so I believe it is buildid specific.

Comment (vtjnash) [reply src/loading.jl:46]: Ensuring that it uses the same .ji file is probably correct then. We might need to examine whether the package hooks should provide that additional information also, to ensure Revise is accurate about it though?

Comment (Keno) [reply src/loading.jl:46]: Alright, good enough for now then.

**Comment by Keno** [comment]
Updated to bump LCU and cherry-pick the additional fixes from #829

**Comment by KristofferC** [comment]
Interesting that the linux testers don't know about LCU 3.0.1.

**Comment by Keno** [comment]
Probably just hadn't propagated yet. Rerunning.

**Comment by Keno** [comment]
@timholy The last issue here is that `ExponentialUtilities` `v1.9.0` is not compatible with all the julia versions we want to test. Can we switch this test to be more self contained and include some minimal in the test code itself rather than depending on some external package?

**Comment by Keno** [comment]
That's mostly a separate change though. I would propose merging this first and then addressing that last issue.

**Comment by timholy** [comment]
> Can we switch this test to be more self contained and include some minimal in the test code itself rather than depending on some external package?

Yes, that needs to happen. I have been doing that for newer tests, but I haven't migrated old tests. Doesn't need to stand in the way of this PR.

**Review by timholy — APPROVED** [review]

**Pull request merged**